src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMGenerator.cs

```diff
@@ -0,0 +1,88 @@
+            : base(targetFrameworkMoniker, cliPath, packagesPath, runtimeFrameworkVersion: null)
+        {
+            CustomRuntimePack = customRuntimePack;
+            AotCompilerPath = aotCompilerPath;
+            AotCompilerMode = aotCompilerMode;
+        }
+
+        protected override void GenerateProject(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
+        {
+            BenchmarkCase benchmark = buildPartition.RepresentativeBenchmarkCase;
+            var projectFile = GetProjectFilePath(benchmark.Descriptor.Type, logger);
+
+            string useLLVM = "false";
+            if (AotCompilerMode == "llvm")
+            {
+                useLLVM = "true";
+            }
+
+            using (var file = new StreamReader(File.OpenRead(projectFile.FullName)))
+            {
+                var (customProperties, sdkName) = GetSettingsThatNeedsToBeCopied(file, projectFile);
+
+                string content = new StringBuilder(ResourceHelper.LoadTemplate("MonoAOTLLVMCsProj.txt"))
+                    .Replace("$PLATFORM$", buildPartition.Platform.ToConfig())
+                    .Replace("$CODEFILENAME$", Path.GetFileName(artifactsPaths.ProgramCodePath))
+                    .Replace("$CSPROJPATH$", projectFile.FullName)
+                    .Replace("$TFM$", TargetFrameworkMoniker)
+                    .Replace("$PROGRAMNAME$", artifactsPaths.ProgramName)
+                    .Replace("$COPIEDSETTINGS$", customProperties)
+                    .Replace("$CONFIGURATIONNAME$", buildPartition.BuildConfiguration)
+                    .Replace("$SDKNAME$", sdkName)
+                    .Replace("$RUNTIMEPACK$", CustomRuntimePack ?? "")
+                    .Replace("$COMPILERBINARYPATH$", AotCompilerPath)
+                    .Replace("$RUNTIMEIDENTIFIER$", buildRuntimeIdentifier())
+                    .Replace("$USELLVM$", useLLVM)
+                    .ToString();
+
+                File.WriteAllText(artifactsPaths.ProjectFilePath, content);
+            }
+        }
+
+        private string buildRuntimeIdentifier()
+        {
+            string os = "";
+
+            if (RuntimeInformation.IsLinux())
+            {
+                os = "linux";
+            }
+            else if (RuntimeInformation.IsMacOSX())
+            {
+                os = "osx";
+            }
+            else if (RuntimeInformation.IsWindows())
+            {
+                os = "win";
+            }
+
+            return $"{os}-{RuntimeEnvironment.RuntimeArchitecture}";
+        }
+
+        protected override string GetExecutablePath(string binariesDirectoryPath, string programName) => Path.Combine(binariesDirectoryPath, programName);
+
+        protected override string GetBinariesDirectoryPath(string buildArtifactsDirectoryPath, string configuration)
+            => Path.Combine(buildArtifactsDirectoryPath, "bin", TargetFrameworkMoniker, buildRuntimeIdentifier(), "publish");
```

Comment:
the new builder calls `dotnet build`, not `dotnet publish` (`dotnetCliBuilder.Build`). Are you sure that the output will be in the publish folder?

Reply:
I think since I have done this with the AotApp Target:

`<Target Name="AotApp" AfterTargets="Build" DependsOnTargets="Publish">`

It will always run after build, and force a publish.

Reply:
thanks for the explanation!